src/unit_test/test_file_validator.py

```diff
@@ -0,0 +1,461 @@
+        self.validator.log.error.assert_not_called()
+
+    def test_validate_file_name_empty_name(self):
+        """Test validation fails when file name is empty"""
+        self.validator.manifest_rows = [
+            {'file_name': '', 'md5sum': 'abc123'},
+        ]
+        self.validator.configs[FILE_NAME_FIELD] = 'file_name'
+        self.validator.configs[FILE_MD5_FIELD] = 'md5sum'
+        
+        result = self.validator.validate_file_name()
+        
+        self.assertFalse(result, "Should return False for empty file name")
+        self.validator.log.error.assert_called()
+        error_msg = self.validator.log.error.call_args[0][0]
+        self.assertIn("File name is empty", error_msg)
+        self.assertIn("Line 2", error_msg)
+
+    def test_validate_file_name_whitespace_only(self):
+        """Test validation fails when file name contains only whitespace"""
+        self.validator.manifest_rows = [
+            {'file_name': '   ', 'md5sum': 'abc123'},
+        ]
+        self.validator.configs[FILE_NAME_FIELD] = 'file_name'
+        self.validator.configs[FILE_MD5_FIELD] = 'md5sum'
+        
+        result = self.validator.validate_file_name()
+        
+        self.assertFalse(result, "Should return False for whitespace-only file name")
+        error_msg = self.validator.log.error.call_args[0][0]
+        self.assertIn("File name is empty", error_msg)
+
+    def test_validate_file_name_duplicate_same_md5(self):
+        """Test validation passes when duplicate file names have same md5"""
+        self.validator.manifest_rows = [
+            {'file_name': 'file1.txt', 'md5sum': 'abc123'},
+            {'file_name': 'file1.txt', 'md5sum': 'abc123'},  # Same name, same md5
+        ]
+        self.validator.configs[FILE_NAME_FIELD] = 'file_name'
+        self.validator.configs[FILE_MD5_FIELD] = 'md5sum'
+        
+        result = self.validator.validate_file_name()
+        
+        self.assertTrue(result, "Should pass when duplicate names have same md5")
+
+    def test_validate_file_name_duplicate_different_md5(self):
+        """Test validation fails when file name is not unique (different md5s)"""
+        self.validator.manifest_rows = [
+            {'file_name': 'file1.txt', 'md5sum': 'abc123'},
+            {'file_name': 'file1.txt', 'md5sum': 'def456'},  # Same name, different md5
+        ]
+        self.validator.configs[FILE_NAME_FIELD] = 'file_name'
+        self.validator.configs[FILE_MD5_FIELD] = 'md5sum'
+        
+        result = self.validator.validate_file_name()
+        
+        self.assertFalse(result, "Should return False for non-unique file names with different md5s")
+        error_msg = self.validator.log.error.call_args[0][0]
+        self.assertIn("is not unique", error_msg)
+        self.assertIn("Line 3", error_msg)
+
+    def test_validate_file_name_absolute_path_unix(self):
+        """Test validation fails for absolute Unix path"""
```

Comment:
Do we need to fail this? If you copy a file's path from Finder (Mac), as I usually do, it will give you absolute path.